mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+            By default ``None``.
+        calib_size : Optional[float]
+            If X_calib and y_calib are not defined, then the calibration
+            dataset is created with the split defined by calib_size.
+        random_state : int, RandomState instance or ``None``, default=None
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            Controls the shuffling applied to the data before applying the
+            split.
+            Pass an int for reproducible output across multiple function calls.
+            See :term:`Glossary <random_state>`.
+        shuffle : bool, default=True
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            Whether or not to shuffle the data before splitting.
+            If shuffle=False
+            then stratify must be ``None``.
+        stratify : array-like, default=None
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            If not ``None``, data is split in a stratified fashion, using this
+            as the class labels.
+            Read more in the :ref:`User Guide <stratification>`.
+
+        Returns
+        -------
+        MapieCalibrator
+            The model itself.
+        """
+        self._check_method()
+        cv = check_cv(self.cv)
+        estimator = check_estimator_classification(X, y, cv, self.estimator)
+        calibrator = self._get_calibrator(self.calibrator)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        assert type_of_target(y) in ["multiclass", "binary"]
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        random_state = check_random_state(random_state)
+
+        if cv == "prefit":
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.n_classes_ = len(self.classes_)
+            self.calibrators = self._fit_calibrators(
+                X, y, sample_weight, calibrator
+            )
+        if cv == "split":
+            results = get_calib_set(
+                X,
+                y,
+                sample_weight=sample_weight,
+                calib_size=calib_size,
+                random_state=random_state,
+                shuffle=shuffle,
+                stratify=stratify,
+            )
+            X_train, y_train, X_calib, y_calib, sw_train, sw_calib = results
+            X_train, y_train = indexable(X_train, y_train)
+            y_train = _check_y(y_train)
+            sw_train, X_train, y_train = check_null_weight(
+                sw_train,
+                X_train,
+                y_train
+            )
+            self.n_classes_ = len(np.unique(y_train))
+            self.estimator = fit_estimator(
```

Comment:
Why don't you use parallel processing with joblib, as in  MapieRegressor

Reply:
I don't think it's required for the estimator. But maybe we could indeed use it for the calibrator, let's discuss this tomorrow during our code review.